File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlUnaryExpr.cs
namespace T1.SqlSharp.Expressions;

public class SqlUnaryExpr : ISqlExpression
{
    public SqlType SqlType { get; set; } = SqlType.UnaryExpression;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_UnaryExpr(this);
    }

    public UnaryOperator Operator { get; set; } = UnaryOperator.BitwiseNot;
    public required ISqlExpression Operand { get; set; }

    public string ToSql()
    {
        return $"{Operator.ToSql()} {Operand.ToSql()}";
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlUnaryExpr.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression } from './ISqlExpression';

export enum UnaryOperator {
    BitwiseNot
}

export class SqlUnaryExpr implements ISqlExpression {
    SqlType: SqlType = SqlType.UnaryExpression;
    Span: TextSpan = new TextSpan();
    Operator: UnaryOperator = UnaryOperator.BitwiseNot;
    Operand!: ISqlExpression;

    Accept(visitor: any): void {
        // 簡單實作
    }

    ToSql(): string {
        switch (this.Operator) {
            case UnaryOperator.BitwiseNot:
                return `~${this.Operand.ToSql()}`;
            default:
                return this.Operand.ToSql();
        }
    }
} 